src/tracks/gosling-track/gosling-track-model.ts

```diff
@@ -77,11 +76,10 @@ export class GoslingTrackModel {
 
         this.theme = theme ?? getTheme();
 
-        this.dataOriginal = JSON.parse(JSON.stringify(data));
-        this.dataAggregated = JSON.parse(JSON.stringify(data));
+        this.dataAggregated = data; // this will be updated after validity of the spec is checked
 
-        this.specOriginal = JSON.parse(JSON.stringify(spec));
-        this.specComplete = JSON.parse(JSON.stringify(spec));
+        this.specOriginal = spec;
```

Comment:
specOrigional never gets modified so we don't need to make a clone of it

Reply:
Nice catch!